backend/mora/service/address_handler/dar.py

```diff
@@ -0,0 +1,127 @@
+#
+# Copyright (c) 2017-2018, Magenta ApS
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+import flask
+import requests
+
+from . import base
+from ... import mapping
+
+session = requests.Session()
+session.headers = {
+    'User-Agent': 'MORA',
+}
+
+NOT_FOUND = "Ukendt"
+
+
+class DARAddressHandler(base.AddressHandler):
+    scope = 'DAR'
+    prefix = 'urn:dar:'
+
+    def __init__(self, value):
+        super().__init__(value)
+
+        try:
+            self.address_object = self._fetch_from_dar(value)
+        except Exception as exc:
+            self.address_object = {
+                mapping.NAME: NOT_FOUND,
+                mapping.HREF: None,
+                mapping.VALUE: value,
+                mapping.ERROR: str(exc),
+            }
+
+            flask.current_app.logger.warn(
+                'ADDRESS LOOKUP FAILED in {!r}:\n{}'.format(
+                    flask.request.url,
+                    value,
+                ),
+            )
+
+    @property
+    def name(self):
+        return ''.join(self._address_string_chunks(self.address_object))
+
+    @property
+    def href(self):
+        href = (
+            'https://www.openstreetmap.org/'
+            '?mlon={x}&mlat={y}&zoom=16'.format(**self.address_object)
+            if 'x' in self.address_object and 'y' in self.address_object
+            else None
+        )
+
+        return href
+
+    def get_mo_address_and_properties(self):
+        if self.address_object.get(mapping.ERROR):
+            return self.address_object
+        else:
+            return super().get_mo_address_and_properties()
```

Comment:
This all feels a bit too magical; optionally calling `super()`…

Reply:
For the most part, this is to mimic the original implementation, returning an error object rather than the address object in case an error occurred.

Reply:
> For the most part, this is to mimic the original implementation, returning an error object rather than the address object in case an error occurred.

Ah, okay. Perhaps stashing them on separate fields, then?

```python
        if self.error is not None:
            return self.error
        else:
            return super().get_mo_address_and_properties()
```

…or just a slightly cleaner an more readable version of the original.

```python
        if mapping.ERROR in self.address_object:
            return self.address_object
        else:
            return super().get_mo_address_and_properties()
```

Reply:
I'll save it on a separate field.